tests/analyse.py

```diff
@@ -40,8 +40,8 @@ def analyze_mining(deployment,last_layer, layers_per_epoch, layer_avg_size, tota
 
     # assert that each node has created layer_avg/number_of_nodes
     for node in blockmap:
-        first_epoch_blocks = sum([len(blockmap[node]["layers"][str(x)]) for x in range(layers_per_epoch) if str(x) in blockmap[node]["layers"]])
+        blocks_in_relevant_layers = sum([len(blockmap[node]["layers"][str(x)]) for x in range(layers_per_epoch, last_layer) if str(x) in blockmap[node]["layers"]])
```

Comment:
Consider renaming last_layer to num_of_layers to avoid such misuse again.

Reply:
not sure I understand... what's the difference?

Reply:
in the test, the last layer we completed consensus on is 8, the number of layers is 9 (0-8) but last_layer=9 which is confusing...

Reply:
I see, I didn't see it that way - I saw it as "the last layer that started", which is 9.

I've already merged this, but if you feel it makes it clearer you can change it in one of your PRs. Both seem equally clear to me, so I don't mind either way.